pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractTopic.java

```diff
@@ -545,8 +545,11 @@ public void recordAddLatency(long latency, TimeUnit unit) {
 
     protected void setSchemaCompatibilityStrategy (Policies policies) {
         if (policies.schema_compatibility_strategy == SchemaCompatibilityStrategy.UNDEFINED) {
-            schemaCompatibilityStrategy = SchemaCompatibilityStrategy.fromAutoUpdatePolicy(
-                    policies.schema_auto_update_compatibility_strategy);
+            schemaCompatibilityStrategy = SchemaCompatibilityStrategy.valueOf(brokerService.pulsar()
+                    .getConfig().getSchemaCompatibilityStrategy());
+            if (schemaCompatibilityStrategy == SchemaCompatibilityStrategy.UNDEFINED) {
```

Comment:
Could we use the equals method of the enum `SchemaCompatibilityStrategy.UNDEFINED`.

such as
```
SchemaCompatibilityStrategy.UNDEFINED.equals(schemaCompatibilityStrategy)
```

Reply:
Because it compares enums, we don't need to use equals

Reply:
Ok